### PR TITLE
Remove safe mode

### PR DESCRIPTION
**Pull request by achow101**
Rationale:

Safe mode is useless. It only disables some RPC commands when large work forks are detected. Nothing else is affected by safe mode. It seems that very few people would be affected by safe mode. The people who use Core as a wallet are primarily using it through the GUI, which safe mode does not effect. In the GUI, transactions will still be made as normal; only a warning is displayed. 

I also don't think that we should be disabling RPC commands or any functionality in general. If we do, it should be done consistently, which safe mode is not. If we want to keep the idea of a safe mode around, I think that the current system needs to go first before a new system can be implemented.

### CODE REVIEW

**Comment by gmaxwell** [comment]
Hm! It's not clear if we should be removing this, or making it better, or removing it as a first step in making it better.

I think it's desirable to have your wallet be able to fail-safe when it has detected something awful is going on with the network.  Right now this ability is underused and inconsistent.

**Comment by achow101** [comment]
I should probably be made better, but its current state, IMO, isn't very useful. It only disables a couple of RPC commands in the event of a large fork (and detection of those are apparently broken - see #9443). The wallet itself and the GUI (which most users use) don't actually do anything when safe mode is triggered. All they do is display a warning about the large work fork that it has detected.

**Comment by laanwj** [comment]
You don't provide *any* rationale.
As is, NACK.

**Comment by achow101** [comment]
@laanwj I added some rationale in the first post.

**Comment by laanwj** [comment]
What to do with safe mode should be discussed some time before 0.16.

**Comment by jgarzik** [comment]
concept ACK

**Comment by TheBlueMatt** [comment]
Yea, I really think this should be improved, not removed, but should be discussed in a meeting, likely.

**Comment by achow101** [comment]
rebased

File: src/rpc/blockchain.cpp [hunk 1]
```diff
@@ -1532,35 +1532,35 @@ UniValue getchaintxstats(const JSONRPCRequest& request)
...
-    { "blockchain",         "getmempoolentry",        &getmempoolentry,        true,  {"txid"} },
-    { "blockchain",         "getmempoolinfo",         &getmempoolinfo,         true,  {} },
-    { "blockchain",         "getrawmempool",          &getrawmempool,          true,  {"verbose"} },
-    { "blockchain",         "gettxout",               &gettxout,               true,  {"txid","n","include_mempool"} },
-    { "blockchain",         "gettxoutsetinfo",        &gettxoutsetinfo,        true,  {} },
-    { "blockchain",         "pruneblockchain",        &pruneblockchain,        true,  {"height"} },
-    { "blockchain",         "verifychain",            &verifychain,            true,  {"checklevel","nblocks"} },
-
-    { "blockchain",         "preciousblock",          &preciousblock,          true,  {"blockhash"} },
+{ //  category              name                      actor (function)         argNames
```
Comment (laanwj) [inline src/rpc/blockchain.cpp]: I really like how this cleans up the dispatch tables, so I've filed #11179 which takes that part already.

**Comment by MarcoFalke** [comment]
Needs rebase if still relevant.

**Comment by achow101** [comment]
Rebased (rewritten really).

**Comment by MeshCollider** [comment]
utACK https://github.com/bitcoin/bitcoin/commit/809519da9a912e8a1b7313d433a646f9fc371086

**Comment by laanwj** [comment]
Probably we should just push forward with this for 0.17.0.

**Comment by laanwj** [comment]
Opened rebased version as #13090

**Pull request closed without merging**